docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,131 @@
+
+The attributes are organized in alphabetical order.
+
+## `@_alignment`
+
+## `@_borrowed`
+
+## `@_cdecl`
+
+Similar to `@_silgen_name` but uses the C calling convention.
+This attribute doesn't have very well-defined semantics.
+
+For potential ideas on stabilization, see
+[Formalizing `@cdecl`](https://forums.swift.org/t/formalizing-cdecl/40677).
+
+## `@_dynamicReplacement(for:)`
+
+Marks a function as the dynamic replacement for another `dynamic` function.
+This is similar to method swizzling in other languages such as Objective-C,
+except that the replacement happens at program start (or loading a shared
+library), instead of at an arbitrary point in time.
+
+For more details, see [Dynamic method replacement](https://forums.swift.org/t/dynamic-method-replacement/16619).
+
+## `@_effects`
+
+## `@_exported`
+
+Use to mark an imported module to re-export all its declarations.
+This attribute is most commonly used by overlays.
+
+```
+// module M
+func f() {}
+
+// module N
+@_exported import M
+
+// module P
+import N
+let () = N.f() // OK
+```
+
+## `@_fixed_layout`
+
+## `@_hasInitialValue`
+
+## `@_hasMissingDesignatedInitializers`
+
+## `@_hasStorage`
+
+## `@_implementationOnly`
+
+Used to mark an imported module as an implementation detail.
+This prevents types from that module being exposed in API
+(types of public functions, constraints in public extension etc.)
+and ABI (usage in `@inlinable` code).
+
+## `@_implements`
+
+## `@_inheritsConvenienceInitializers`
+
+## `@_nonEphemeral`
+
+## `@_nonoverride`
+
+## `@_optimize(...)`
+
+## `@_originallyDefinedIn`
+
+Marks a declaration as being originally defined in a different module,
+changing the name mangling. This can be used to move declarations
+from a module to one of the modules it imports without breaking clients.
+
+## `@_private(sourceFile:)`
+
+Fully bypasses access control, allowing access to private declarations
+in the imported module. The imported module needs to be compiled with
+`-Xfrontend -enable-private-imports` for this to work.
+
+## `@_semantics`
+
+Marks a function as having particular high-level semantics that are
+specially recognized by the SIL optimizer.
+
+## `@_silgen_name`
+
+Changes the symbol name for a function, similar to an ASM label in C,
+except that the platform symbol mangling (leading underscore on Darwin)
+is maintained.
+
+Since this has label-like behavior, it may not correspond to any declaration;
+if so, it is assumed that the function is implemented in C.
+
+A function defined by `@_silgen_name` is assumed to use the Swift ABI.
+
+## `@_specialize(...)`
+
+Forces generation of a specialized implementation for a generic declaration.
+
+See [docs/Generics.rst][] for more details.
+
+## `@_spi(spiName)`
+
+Marks a declaration as SPI (System Programming Interface),
+instead of API. Modules exposing SPI and using library evolution
+generate an additional `.private.swiftinterface` file in addition
+to the usual `.swiftinterface` file.
```

Comment:
IIRC, you need to actually pass a separate `-emit-private-module-interface-path`; otherwise it won't create the second file.